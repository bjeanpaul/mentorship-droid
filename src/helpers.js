import { find, isUndefined, partialRight as partial } from 'lodash';
import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const omitNulls = d => omitBy(d, isNull);


export const trap = (type, fn) => v => v instanceof type
  ? Promise.resolve(v).then(fn)
  : Promise.reject(v);


export const switchError = cases => (e, ...xargs) => {
  const [type, fn] = find(cases, ([type, fn]) => e instanceof type) || [];
  return !isUndefined(fn)
    ? Promise.resolve().then(() => fn(e, ...xargs))
    : Promise.reject(e);
};


export const apiAction = ({
  method,
  request,
  success,
  failures,
}) => (...args) => (dispatch, { auth }) => Promise.resolve()
  .then(() => request(...args))
  .then(dispatch)
  .then(() => method(...args, auth))
  .then(partial(success, ...args), partial(switchError(failures), ...args))
  .then(dispatch);


export const staticAction = type => () => ({
  type
});


export const dataAction = type => data => ({
  type,
  payload: data
});


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
