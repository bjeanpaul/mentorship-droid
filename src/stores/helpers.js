import moment from 'moment';
import { values, isUndefined, sortBy, filter, first } from 'lodash';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


// TODO sort?
export const getCategories = ({ entities: { categories } }) => values(categories);


// TODO sort?
export const getCategoryActivities = ({
  entities: { activities },
}, targetCategoryId) => (
  values(activities)
    .filter(({ category }) => category === targetCategoryId));


export const getCategory = ({ entities: { categories } }, id) => categories[id];


export const getActivity = ({ entities: { activities } }, id) => activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) => values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) => scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall && scheduledCall.activity && getActivity(state, scheduledCall.activity);
};


export const getEvents = ({ entities: { events } }) => values(events);


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) => (
  filter(callNotes, { callActivity: activityId }));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => callTime);
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};
