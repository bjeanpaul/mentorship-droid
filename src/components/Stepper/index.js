import { range, clamp, fromPairs } from 'lodash';
import React, { Component, PropTypes } from 'react';
import BackAndroid from 'BackAndroid';

import { ProgressBar, BaseView, FormView, NavigationStack } from 'src/components';
import { createRoute } from 'src/navigationHelpers';


const Step = ({
  children,
}) => (
  <BaseView>
    {children}
  </BaseView>
);


class Stepper extends Component {
  constructor(props) {
    super(props);

    this.state = {
      index: 0,
    };

    this.onNextPress = this.onNextPress.bind(this);
    this.onBackPress = this.onBackPress.bind(this);
    this.onNativeBackPress = this.onNativeBackPress.bind(this);
  }

  componentDidMount() {
    BackAndroid.addEventListener('hardwareBackPress', this.onNativeBackPress);
  }

  componentWillUnmount() {
    BackAndroid.removeEventListener('hardwareBackPress', this.onNativeBackPress);
  }

  onNativeBackPress() {
    if (this.state.index > 0) {
      this.onBackPress();
      return true;
    } else {
      return false;
    }
  }

  onNextPress() {
    this.incrStep(+1);
  }

  onBackPress() {
    this.incrStep(-1);
  }

  getRoutes() {
    const routes = React.Children.toArray(this.props.children)
      .map((child, i) => [i, this.createRoute(child)]);

    return fromPairs(routes);
  }

  getNumSteps() {
    return React.Children.count(this.props.children);
  }

  getNavigationState() {
    return {
      index: this.state.index,
      routes: range(this.getNumSteps()).map(i => createRoute(i)),
    };
  }

  createRoute(el) {
    return React.cloneElement(el, {
      onNextPress: this.onNextPress,
      onBackPress: this.onBackPress,
    });
  }

  incrStep(v) {
    this.setState({
      index: clamp(this.state.index + v, 0, this.getNumSteps() - 1),
    });
  }

  render() {
    const numSteps = React.Children.count(this.props.children);
    const completed = (this.state.index + 1) / numSteps;

    return (
      <FormView>
        {!this.props.hideProgress && <ProgressBar completed={completed} />}
        <NavigationStack
          navigationState={this.getNavigationState()}
          routes={this.getRoutes()}
        />
      </FormView>
    );
  }
}


Stepper.propTypes = {
  hideProgress: PropTypes.bool,
  children: PropTypes.arrayOf(Step).isRequired,
};


Step.propTypes = {
  children: PropTypes.any.isRequired,
};


export {
  Stepper as default,
  Step,
};
