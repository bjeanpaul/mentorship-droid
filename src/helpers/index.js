import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const conj = (...args) => Object.assign({}, ...args);


export const omitNulls = d => omitBy(d, isNull);


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
