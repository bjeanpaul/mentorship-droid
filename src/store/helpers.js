import moment from 'moment';
import { values, isUndefined, sortBy, filter, first, orderBy } from 'lodash';

import { getCurrent } from 'src/navigationHelpers';
import { TYPES_TO_COLLECTIONS } from 'src/constants/events';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


export const getCategories = ({ entities }) =>
  sortBy(values(entities.categories), 'ordinal');


export const getCategoryActivities = ({ entities }, targetCategoryId) => {
  const activities = values(entities.activities)
    .filter(({ category }) => category === targetCategoryId);

  return sortBy(activities, 'ordinal');
};


export const getCategory = ({ entities: { categories } }, id) =>
  categories[id];


export const getActivity = ({ entities: { activities } }, id) =>
  activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) =>
  values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) =>
  scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall
      && scheduledCall.activity
      && getActivity(state, scheduledCall.activity);
};


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) =>
  values(callNotes)
    .filter(({ callActivityId }) => callActivityId === activityId);


export const getCallNotes = ({
  entities: { callNotes },
}) =>
  sortBy(callNotes, ({ callStartTime }) => +moment(callStartTime));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => +moment(callTime));
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};


export const getScheduledCallsBetween = (state, start, end, time = Date.now()) => {
  // for some reason, moment durations are behaving inconsistently in tests,
  // thus the lack of moment usage here
  const startTime = +moment(time) - start;
  const endTime = +moment(time) + end;

  return getScheduledCalls(state)
    .filter(({ callTime }) => {
      const t = +moment(callTime);
      return startTime <= t && t <= endTime;
    });
};


const getEventObject = (state, event) => {
  const collection = state.entities[TYPES_TO_COLLECTIONS[event.eventType]];
  return collection && collection[event.objectId];
};


// TODO return the associated entitites to make the containers' jobs easier
export const getEvents = (state) =>
  values(state.entities.events)
    .filter(event => getEventObject(state, event));


export const getMessages = ({ entities }) => []
  .concat(values(entities.messages))
  .concat(values(entities.pendingMessages));


export const getMessage = ({ entities }, id) => entities.messages[id];


export const getActiveTopRoute = store => getCurrent(store.navigation.top);


export const getBlogPost = ({ entities }, id) => entities.blogPosts[id];


export const getBlogPosts = ({ entities }) =>
  orderBy(entities.blogPosts, [({ createdAt }) => +moment(createdAt)], ['desc']);
