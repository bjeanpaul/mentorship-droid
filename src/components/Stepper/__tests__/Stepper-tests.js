import React from 'react';
import { Text } from 'react-native';
import Stepper, { Step } from 'src/components/Stepper';
import { uidEquals } from 'app/scripts/helpers';


describe('Stepper', () => {
  const createComponent = (props = {}) => (
    <Stepper {...props}>
      <Step uid="step:1"><Text>Step 1</Text></Step>
      <Step uid="step:2"><Text>Step 2</Text></Step>
    </Stepper>
  );

  it('should render its steps', () => {
    expect(render(createComponent())).toMatchSnapshot();
  });

  it('should toggle the rendering of the progressBar', () => {
    expect(render(createComponent({
      hideProgress: true,
    }))).toMatchSnapshot();
  });

  it('should page between steps when page buttons are pressed', () => {
    const el = shallow(createComponent());
    expect(el.state().index).toEqual(0);

    el.find('NavigationStack')
      .shallow()
      .findWhere(uidEquals('step:1'))
      .simulate('nextPress');

    expect(el.state().index).toEqual(1);

    el.find('NavigationStack')
      .shallow()
      .findWhere(uidEquals('step:2'))
      .simulate('backPress');

    expect(el.state().index).toEqual(0);
  });

  it('should clamp the step index betwen the range of steps', () => {
    const el = shallow(createComponent());
    expect(el.state().index).toEqual(0);

    el.find('NavigationStack')
      .shallow()
      .findWhere(uidEquals('step:1'))
      .simulate('backPress');

    expect(el.state().index).toEqual(0);

    el.find('NavigationStack')
      .shallow()
      .findWhere(uidEquals('step:1'))
      .simulate('nextPress');

    el.find('NavigationStack')
      .shallow()
      .findWhere(uidEquals('step:2'))
      .simulate('nextPress');

    expect(el.state().index).toEqual(1);
  });

  describe('onNativeBackPress', () => {
    it('should go back', () => {
      const el = shallow(createComponent());

      el.setState({ index: 1 })
        .instance()
        .onNativeBackPress();

      expect(el.state().index).toEqual(0);
    });

    it('should return false when at step 0', () => {
      let res;
      const el = shallow(createComponent());

      res = el.setState({ index: 1 })
        .instance()
        .onNativeBackPress();

      expect(res).toBe(true);

      res = el.setState({ index: 0 })
        .instance()
        .onNativeBackPress();

      expect(res).toBe(false);
    });
  });
});
