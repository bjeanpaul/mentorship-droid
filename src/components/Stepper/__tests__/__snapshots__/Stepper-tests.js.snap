exports[`Stepper should map props correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Object {
        "backgroundColor": "#e5e5e5",
        "flexDirection": "row",
        "height": 6
      }
    }>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#f77040"
          },
          Object {
            "flex": 1
          }
        ]
      } />
    <View
      style={
        Object {
          "flex": 0
        }
      } />
  </View>
  <View
    onLayout={[Function bound _onLayout]}
    style={
      Array [
        Object {
          "flex": 1
        },
        undefined
      ]
    }>
    <View
      style={
        Object {
          "flex": 1
        }
      }>
      <View
        style={
          Object {
            "flex": 1
          }
        }>
        <View
          onMoveShouldSetResponder={[Function onMoveShouldSetResponder]}
          onMoveShouldSetResponderCapture={[Function onMoveShouldSetResponderCapture]}
          onResponderEnd={[Function onResponderEnd]}
          onResponderGrant={[Function onResponderGrant]}
          onResponderMove={[Function onResponderMove]}
          onResponderReject={[Function onResponderReject]}
          onResponderRelease={[Function onResponderRelease]}
          onResponderStart={[Function onResponderStart]}
          onResponderTerminate={[Function onResponderTerminate]}
          onResponderTerminationRequest={[Function onResponderTerminationRequest]}
          onStartShouldSetResponder={[Function onStartShouldSetResponder]}
          onStartShouldSetResponderCapture={[Function onStartShouldSetResponderCapture]}
          pointerEvents="none"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 10,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 1000000
                },
                Object {
                  "translateY": 1000000
                }
              ]
            }
          }>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1
                },
                undefined
              ]
            }>
            <Text>
              Step 2
            </Text>
          </View>
        </View>
        <View
          onMoveShouldSetResponder={[Function onMoveShouldSetResponder]}
          onMoveShouldSetResponderCapture={[Function onMoveShouldSetResponderCapture]}
          onResponderEnd={[Function onResponderEnd]}
          onResponderGrant={[Function onResponderGrant]}
          onResponderMove={[Function onResponderMove]}
          onResponderReject={[Function onResponderReject]}
          onResponderRelease={[Function onResponderRelease]}
          onResponderStart={[Function onResponderStart]}
          onResponderTerminate={[Function onResponderTerminate]}
          onResponderTerminationRequest={[Function onResponderTerminationRequest]}
          onStartShouldSetResponder={[Function onStartShouldSetResponder]}
          onStartShouldSetResponderCapture={[Function onStartShouldSetResponderCapture]}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 10,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0
                },
                Object {
                  "translateY": 0
                }
              ]
            }
          }>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1
                },
                undefined
              ]
            }>
            <Text>
              Step 2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Stepper should toggle the rendering of the progressBar 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    onLayout={[Function bound _onLayout]}
    style={
      Array [
        Object {
          "flex": 1
        },
        undefined
      ]
    }>
    <View
      style={
        Object {
          "flex": 1
        }
      }>
      <View
        style={
          Object {
            "flex": 1
          }
        }>
        <View
          onMoveShouldSetResponder={[Function onMoveShouldSetResponder]}
          onMoveShouldSetResponderCapture={[Function onMoveShouldSetResponderCapture]}
          onResponderEnd={[Function onResponderEnd]}
          onResponderGrant={[Function onResponderGrant]}
          onResponderMove={[Function onResponderMove]}
          onResponderReject={[Function onResponderReject]}
          onResponderRelease={[Function onResponderRelease]}
          onResponderStart={[Function onResponderStart]}
          onResponderTerminate={[Function onResponderTerminate]}
          onResponderTerminationRequest={[Function onResponderTerminationRequest]}
          onStartShouldSetResponder={[Function onStartShouldSetResponder]}
          onStartShouldSetResponderCapture={[Function onStartShouldSetResponderCapture]}
          pointerEvents="none"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 10,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 1000000
                },
                Object {
                  "translateY": 1000000
                }
              ]
            }
          }>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1
                },
                undefined
              ]
            }>
            <Text>
              Step 2
            </Text>
          </View>
        </View>
        <View
          onMoveShouldSetResponder={[Function onMoveShouldSetResponder]}
          onMoveShouldSetResponderCapture={[Function onMoveShouldSetResponderCapture]}
          onResponderEnd={[Function onResponderEnd]}
          onResponderGrant={[Function onResponderGrant]}
          onResponderMove={[Function onResponderMove]}
          onResponderReject={[Function onResponderReject]}
          onResponderRelease={[Function onResponderRelease]}
          onResponderStart={[Function onResponderStart]}
          onResponderTerminate={[Function onResponderTerminate]}
          onResponderTerminationRequest={[Function onResponderTerminationRequest]}
          onStartShouldSetResponder={[Function onStartShouldSetResponder]}
          onStartShouldSetResponderCapture={[Function onStartShouldSetResponderCapture]}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 10,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0
                },
                Object {
                  "translateY": 0
                }
              ]
            }
          }>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1
                },
                undefined
              ]
            }>
            <Text>
              Step 2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
